### ISSUE TITLE
Adding General Checkpoint Saving and Loading for Training

### ISSUE THREAD

**Issue by jaycehaoranwang** [lolzballs/mie429#4]
Currently the training pipeline only saves the model params and will only allow loading back the model for inference/evaluation, currently we cannot reload the model params and continue training as optimizer variables / loss and other training metrics are not saved. Integrating the ability to save general training checkpoints that can be used to reload and continue training at the checkpoint could be very useful in the future. 

Info on how to do it here: https://pytorch.org/tutorials/beginner/saving_loading_models.html
